benchmarks/tools/benchmark.py

```diff
@@ -0,0 +1,177 @@
+from .utils import bench_print
+
+from argparse import ArgumentParser
+from collections import OrderedDict
+from collections.abc import Iterable
+from itertools import product
+from datetime import datetime
+from os import path, makedirs
+
+import json
+
+
+__all__ = ['Benchmark']
+
+
+def get_argparsers(**kwargs):
+    """
+    Utility that creates the root arguemnt parser and the sub-parser for each mode.
+    """
+    parser = ArgumentParser(**kwargs)
+    subparsers = parser.add_subparsers(dest='mode', help="Mode of operation")
+    help = 'Perform benchmarking runs on target machine'
+    parser_bench = subparsers.add_parser('bench', help=help)
+    help = 'Plot diagrams from stored results'
+    parser_plot = subparsers.add_parser('plot', help=help)
+    return parser, parser_bench, parser_plot
+
+
+class Benchmark(object):
+    """
+    Class storing performance data for a set of benchmark runs
```

Comment:
nitpicking
in devito, docstrings never start with "class ...". 
Rather: "Performance data for a set of benchmark runs ...."